resources/benchmark-runner.mjs

```diff
@@ -41,43 +41,71 @@ class BenchmarkState {
         return !this._testIndex;
     }
 
-    prepareCurrentSuite(runner, frame) {
+    async prepareCurrentSuite(frame, page) {
         const suite = this.currentSuite();
-        return new Promise((resolve) => {
-            frame.onload = () => {
-                suite.prepare(runner, frame.contentWindow, frame.contentDocument).then(resolve);
+        await new Promise((resolve) => {
+            frame.onload = async () => {
+                await suite.prepare(page);
+                resolve();
             }
             frame.src = 'resources/' + suite.url;
         });
     }
 }
 
-export class BenchmarkRunner {
-    constructor(suites, client) {
-        this._suites = suites;
-        this._prepareReturnValue = null;
-        this._client = client;
+const ENTER_KEY_CODE = 13;
+
+class Page {
+    constructor(frame) {
+        this._frame = frame;
     }
 
-    waitForElement(selector) {
+    async waitForElement(selector) {
         return new Promise((resolve) => {
-            const contentDocument = this._frame.contentDocument;
-
-            function resolveIfReady() {
-                const element = contentDocument.querySelector(selector);
-                if (element) {
-                    window.requestAnimationFrame(function () {
-                        return resolve(element);
-                    });
-                    return;
-                }
+          const resolveIfReady = () => {
+            const element = this.querySelector(selector);
+            if (element) {
+                window.requestAnimationFrame(() => {
+                    return resolve(element);
+                });
+            } else {
                 setTimeout(resolveIfReady, 50);
             }
-
-            resolveIfReady();
+          };
+          resolveIfReady();
         });
     }
 
+    querySelector(selector) {
```

Comment:
WebKit style is to put { in a new line.
I know we haven't been consistent but we might as well as in new code.